libvast/src/logical_pipeline.cpp

```diff
@@ -281,6 +281,34 @@ auto make_local_producer(std::span<logical_operator_ptr> ops,
 
 } // namespace
 
+logical_pipeline::logical_pipeline(const logical_pipeline& other) {
+  if (this != &other) {
+    ops_.clear();
+    ops_.reserve(other.ops_.size());
+    for (const auto& op : ops_)
+      ops_.push_back(op ? op->copy() : nullptr);
+  }
+}
+
+logical_pipeline& logical_pipeline::operator=(const logical_pipeline& rhs) {
+  if (this != &rhs) {
+    ops_.clear();
+    ops_.reserve(rhs.ops_.size());
+    for (const auto& op : ops_)
+      ops_.push_back(op ? op->copy() : nullptr);
+  }
+  return *this;
+}
```

Comment:
This violates our invariant that a `logical_pipeline` does not contain another `logical_pipeline`. Furthermore, we currently prohibit `nullptr` operators in `make`, and I would personally prefer to maintain this invariant everywhere. We can perhaps throw an exception if `copy()` returns a `nullptr`.

Reply:
Good catch, I'll simplify the logic and flatten here.